Fix #35: Accept for commands in "/command@bot" format

RedL0tus opened the pull request:
Fixes #35 .

It seems can work... I've tested it with the simplest reply bot.

What I've done is fetch the username through `get_me()` and when the command received ends with the username, remove the username & `@` from the command then process it as normal.

I've just leant Rust for several weeks so I may not using the best way 😂

RedL0tus committed:
Fetch username while starting the bot

RedL0tus committed:
Remove bot's username while processing commands

bytesnake commented on src/bot.rs:
```diff
@@ -340,6 +350,7 @@ impl RcBot {
 
     /// helper function to start the event loop
     pub fn run<'a>(&'a self, core: &mut Core) -> Result<(), Error> {
+        *self.inner.name.borrow_mut() = Some(core.run(self.get_me().send()).unwrap().1.username.unwrap());
```
Hey, this works for sure. Here is another way which doesn't block before the main run command.
``` rust
// we want to use the get_me function here, therefore import the corresponding trait
use functions::FunctionGetMe;

// create a local copy of the bot to circumvent lifetime issues
let bot = self.inner.clone();

// create a new task which resolves the bot name and then set it in the struct
let resolve_name = self.get_me().send()
    .map(move |user| {
        if let Some(name) = user.1.username {
            bot.name.replace(Some(format!("@{}", name)));                                                                                                                            
        }
    }); 

// spawn the task
self.inner.handle.spawn(resolve_name.map_err(|_| ()));
```

RedL0tus replied:
I've changed to your version in the latest commit.
Seems I have to learn more about futures 😂

bytesnake commented on src/bot.rs:
```diff
@@ -311,8 +314,15 @@ impl RcBot {
+                        if let Some(command) = content.next() {
+                            if command.starts_with("/") {
+                                let cmd = {
+                                    if command.ends_with(&*self.inner.name.borrow().as_ref().unwrap().as_str()) {
```
This will crash when the name is None in Bot. You could also make the cmd mutable and then just cut off the last part: 
``` rust
if cmd.ends_with(name) {
    cmd = cmd.rsplitn(2, '@').skip(1).next().unwrap();
}
```
(this assumes that in name is "@BotName")

RedL0tus replied:
So I added an extra `if let` here to prevent crashing.
(But every bot has a username, isn't it 😂)

bytesnake reviewed (changes requested):
Hey, thanks for the pull request! I added two comments to your changes. To work with futures is sometimes cumbersome ...

RedL0tus committed:
Follow the review comments at #40 ... XD

Sorry, I really don't know how to write this commit message.

I add an extra `if let` before modifying `cmd` to prevent panic by a None.
